docs_src/design/adr/core/0022-UoM.md

```diff
@@ -0,0 +1,199 @@
+
+There are various specifications and standards around unit of measure.  Specifically, there are several options to choose from as it relates to the exchange of data in electronic communications - and units of measure associated in that exchange.  As examples, two big competing standards around EDI (electronic data exchange) that both have associated unit of measure codes are:
+
+- [ANSI X12: EDI standard used mostly in the US](https://ediacademy.com/blog/x12-unit-of-measurement-codes/)
+- [EDIFACT: UN EDI standard used mostly in Europe and Asia](https://unece.org/fileadmin/DAM/cefact/recommendations/rec20/rec20_rev3_Annex2e.pdf)
+
+The [Unified Code for Units of Measure](https://en.wikipedia.org/wiki/Unified_Code_for_Units_of_Measure) provides an alternative list (not a standard) that is used by various organizations like OSGI and the Eclipse Foundation.
+
+While standards exist, use by various open source projects (especially IoT/edge projects) is inconsistent and haphazard.   Groups like oneM2M seem to define their own selection of units in specifications per vertical (home for example) while Kura doesn't even appear to use the UoM JSR (a Java related unit of measure specification for Java applications like Kura).
+
+## Decision
+
+It would be speculative and inappropriate for EdgeX to select a unit of measure standard which is not widely adopted in the industry or choose a static unit of measure list that is incomplete with regard to possible IoT / edge use case needs.  At this time, there does not appear to be a single and unequivocal standard for units of measure that encompasses all EdgeX related use cases (now and in the future). 
+
+Therefore, EdgeX chooses not to select or adopt a unit of measure specification, standard, or code list to apply across the platform.  Instead, EdgeX adopters will be allowed to optionally specify which unit of measure specification, standard, or unit of measure code list they would like used in their instance(s) of EdgeX.
+
+### Specifying the Units of Measure
+
+Units of measure allowed by the instance of EdgeX will be specified in a configuration file (in TOML format).  This can either be specified directly in the core metadata configuration file (but then hard to change when using a containerized or snap form of the application) or in a separate and external configuration TOML file (making it easier for an adopter to provide units of measure even when the application is containerized or snapped up).  
```

Comment:
> @tonyespy
I'm not sure I understand the need to have a global configuration file? If the default core-metadata configuration.toml file includes default UoM values, they can be overridden via environment overrides (applies to docker containers or snaps). Additionally both docker and snaps allow entire configuration.toml files to be provided to overridden entirely (i.e. using docker volumes and/or snap content interfaces).

It is true that passing a local file is possible to both docker containers and installed snaps. The text mentions that it is hard (not that it is impossible). I believe setting the values via env vars is not possible because of the dynamic keys for unit categories, if they don't already exist in the config file - explained on line 101. (This is probably something solvable though)

From the discussions, the support for URIs instead of local file paths was to permit passing either of local and remote files. One use case mentioned for a remote file is to create public unit registries and let the community maintain and use them. For example, the project may provide an online TOML file for [X12 units](https://ediacademy.com/blog/x12-unit-of-measurement-codes/) and the community may create few others.

This would also allow deployments with a custom docker compose script with env vars and no additional files, taking care of a whole deployment and scale out. The same can be done with the snaps. I'm simplifying a bit and ignoring profile and device config files. There was also discussion on supplying those files externally if this turns out to be useful (assuming Cosul can't do the same and supply initial configs centrally?).

Perhaps line 72 should be modified to clarify the uses. It currently says:
> The location of the external UoM file (specified by URI) should point to an accessible website, shared volume, volume mount, etc. in order to allow for the adopter to provide the units of measure independently during configuration/setup of the EdgeX instance without requiring a build of the metadata service or a reconstruction of the Docker image/container.

Edit: I now see further details on line 180.